### PR TITLE
Implement remaining `kind`s and `method:transform(spaces)` snippets for `c99_tc3`

### PR DESCRIPTION
**Pull request by jssblck**
# Overview

Adds implementations for:
- `Kind::Signature`
- `Kind::Body`
- `Transform::Space`

As well:
- Adds the notion of "targets" to the snippet library, of which only `Target::Function` currently exists. This is meant to leave us an escape hatch for later targets without breaking changes, and to make intentions like [this comment](https://github.com/fossas/foundation-libs/pull/14#discussion_r1310686186) more clear.
  - As an example, a future `Target` might be `File` (to unify snippet scanning & VSI).
- The transformed content used to create a snippet fingerprint is now attached to the snippet. This was primarily needed for testing. This will increase memory pressure of snippet generation; if it proves to be too much we can always feature flag this later.
- Simplifies `Transforms` to work like other `FlagSets` (they now have a `Transform::Code` option instead of relying on magical unification).
- Allows `Method::Raw` to be disabled (although this is mainly for testing).

`Transform::Code` and `Transform::Comment` are marked `unimplemented!`, so currently result in a panic if provided. The plan is to fill these in before MVP but we need to move on for now to derisk hitting our targets.

## Acceptance criteria

Users can extract `Signature` and `Body` functions using `Method::Raw` and `Method::Transform(Transform::Space)`.

## Testing plan

Relied on automated testing.

## Risks

The `unimplemented!` behavior is potentially surprising but fine for now, low risk.

## References

Partially implements https://fossa.atlassian.net/browse/ANE-1117

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).

### CODE REVIEW

File: snippets/src/text/normalize_space.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+mod tests {
+    use super::*;
+
+    #[test]
+    fn multi_spaces_compiles() {
+        let _ = MULTI_SPACES;
+    }
+
+    #[test]
+    fn normalizes_spaces() {
```
Comment (spatten) [inline snippets/src/text/normalize_space.rs:24]: [Optional] Can/should we add some windows newlines in a test too?

File: snippets/tests/it/language/testdata/c99_tc3/hello_world_comment.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/* indicates error code */ int /* the classic starting point */ main /* who needs params? */ (/* no params, because let's keep this "simple" */) // and end with a bang
+{ // Some hello world, huh?
+  // what if we could never decide on a comment style?
+
+  /**
+* // Fortunately, GCC itself dies on nested multiline comments
+    *    // So at least we don't have to worry about that...
+  */// But we do need to make sure to handle single line after multiline
+  printf/* () */("hello world\n" /* ) */ ); // the classic beginning
```
Comment (spatten) [inline snippets/tests/it/language/testdata/c99_tc3/hello_world_comment.c:14]: This line made me shudder. Sheesh!

**Review by spatten — APPROVED** [review]
This all looks great! Just one minor comment on a test

File: snippets/src/text/normalize_space.rs (hunk 1 above)
Comment (jssblck) [reply snippets/src/text/normalize_space.rs:24]: It's in! 😁 https://github.com/fossas/foundation-libs/blob/b260c007c08990e80a1f95984b9abb6ebe43d577/snippets/tests/it/language/c99_tc3.rs#L30-L72

**Pull request merged**